### ISSUE TITLE
centos 7 can't add TBF qdisc

### ISSUE THREAD

**Issue by langyxxl** [vishvananda/netlink#251]
I use this qos config :
qdisc_tbf := netlink.Tbf{                                              
        QdiscAttrs: qdisc_attrs,                                            
        Rate:       25000000,                                              
        Limit:      uint32(1250000), 
        Buffer:     uint32(125000),                                        
    }                                                                      
err := nlh.QdiscAdd(&qdisc_tbf)
I got "invalid argument" from centos7.
But in ubuntu 16.04 ,I can run successful.

**Comment by candiLin** [comment]
hi, I have the same problem with you,  do you know how to solve this problem?

**Comment by vishvananda** [comment]
It is possible that your version of the kernel does not support 64bit rates, which means you would need to set the rate lower than u32 max. Does setting the rate lower work properly?

**Comment by langyxxl** [comment]
u32max is 4,294,967,295
Bigger than   25,000,000

**Comment by vishvananda** [comment]
ah you are correct. I misread it as 25,000,000,000

**Comment by candiLin** [comment]
I updated the kernel version to 4.12, it can work.  So maybe it's the difference of th kernel version causes the failure

**Comment by vishvananda** [comment]
Yeah it looks like the older kernel needs a slightly different message. I
will try to make a version that works with the older kernel.

**Comment by wangxf1987** [comment]
I think the code have issue, the QdiscAdd func need to var and the type is Qdisc, we refer to the example and qdisc_test.go, actually, the var type is Tbf.

**Comment by wangxf1987** [comment]
@langyxxl Hi, Can your paste the qdisc_attrs struct ?
